core/lib/json_api_client_extension/validations/remote_uniqueness_validation.rb

```diff
@@ -0,0 +1,18 @@
+module JsonApiClientExtension
+  module Validations
+    class RemoteUniquenessValidator < ::ActiveModel::EachValidator
+      def validate_each(record, attribute, value)
+        list = record.class.where({attribute => value}).paginate(page: 1, per_page: 1).to_a
```

Comment:
can you do `record.class.find_first({attribute => value}).to_a` ?

and `list.any? { |e| e && e.id != record.id  }`